src/main/java/org/cru/cas/client/integration/LogoutBroadcastFilter.java

```diff
@@ -0,0 +1,155 @@
+package org.cru.cas.client.integration;
+
+import static org.cru.cas.client.integration.Util.getRequiredInitParam;
+
+import java.io.BufferedInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.HttpURLConnection;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.util.Arrays;
+import java.util.Enumeration;
+import java.util.Set;
+import java.util.stream.Collectors;
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletInputStream;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A filter to broadcast logout requests to other nodes operating behind a load balancer.
+ *
+ * This behaves more like a servlet than a filter,
+ * but it is a filter instead of a servlet so that it can run before the
+ * {@link org.jasig.cas.client.session.SingleSignOutFilter} filter.
+ */
+public class LogoutBroadcastFilter implements Filter {
```

Comment:
I suppose that neither will this `LogoutBroadcastFilter` be used in the first deploy.

Reply:
That's correct